lib/WUI/nhttp/status_page.h

```diff
@@ -36,7 +41,23 @@ class StatusPage {
         , json_content(json_content) {}
     bool want_read() const { return false; }
     bool want_write() const { return true; }
-    Step step(std::string_view input, bool terminated_by_client, uint8_t *output, size_t output_size);
+    virtual Step step(std::string_view input, bool terminated_by_client, uint8_t *output, size_t output_size);
+};
+
+enum class AuthMethod {
+    ApiKey,
+    Digest
+};
+class UnauthenticatedStatusPage final : public StatusPage {
+private:
+    char nonce[http::NONCE_SIZE];
+    bool nonce_stale;
+    AuthMethod auth_method;
+
+public:
+    UnauthenticatedStatusPage(http::Status status, CloseHandling close_handling, bool json_content, AuthMethod auth_method, bool nonce_stale = false, const char *nonce = "", const char *extra_content = "");
```

Comment:
I'd suggest not to use the default values of parameters.

Is the extra_content used anywhere?